spec/system/admin_customize_themes_spec.rb

```diff
@@ -145,21 +145,100 @@
     end
   end
 
-  # TODO(osama) unskip this test when the "Themes and components" link is
-  # changed to the new config customize page
   context "when visting a theme's page" do
-    xit "has a link to the themes page" do
+    it "has a link to the themes page" do
       visit("/admin/customize/themes/#{theme.id}")
-      expect(admin_customize_themes_page).to have_back_button_to_themes_page
+      expect(theme_page).to have_back_button_to_themes_page
     end
   end
 
   context "when visting a component's page" do
     fab!(:component) { Fabricate(:theme, component: true, name: "Cool component 493") }
 
-    xit "has a link to the components page" do
+    it "has a link to the components page" do
       visit("/admin/customize/themes/#{component.id}")
-      expect(admin_customize_themes_page).to have_back_button_to_components_page
+      expect(theme_page).to have_back_button_to_components_page
+    end
+  end
+
+  describe "theme color palette editor" do
+    before { SiteSetting.use_overhauled_theme_color_palette = true }
+
+    it "shows color palette editor when feature is enabled" do
+      theme_page.visit(theme.id)
+
+      expect(theme_page).to have_color_palette_editor
+      expect(theme_page).to have_no_color_scheme_selector
+    end
+
+    it "doesn't show color palette editor when feature is disabled" do
+      SiteSetting.use_overhauled_theme_color_palette = false
+      theme_page.visit(theme.id)
+
+      expect(theme_page).to have_no_color_palette_editor
+      expect(theme_page).to have_color_scheme_selector
+    end
+
+    it "allows editing colors without affecting other themes" do
```

Comment:
I think the controller test we have for this is enough. We don't really need a system test to verify other records will be affected.